### ISSUE TITLE
训练中文数据集thchs30，应该如何修改yaml配置文件

### ISSUE THREAD

**Issue by q248953144** [PaddlePaddle/Parakeet#40]
是否可以提供下yaml参数的具体说明

**Comment by iclementine** [comment]
请问你是需要了解哪一个模型的 yaml 配置呢？

嗯， 后续版本我们会考虑加上更加友好的说明。

**Comment by q248953144** [comment]
> 请问你是需要了解哪一个模型的 yaml 配置呢？
> 
> 嗯， 后续版本我们会考虑加上更加友好的说明。

例如Parakeet/examples/transformer_tts/configs/ljspeech.yaml和Parakeet/examples/waveflow/configs/waveflow_ljspeech.yaml 这些yaml文件里面的参数，是否可提供说明。如果想训练中文数据集thchs30应该如何修改

**Comment by iclementine** [comment]
# waveflow 的配置文件
valid_size: 16  # 预留作为 validation 的句子数
segment_length: 16000  # 训练时用的音频只是一段（不是整句），16000 表示其 sample point 数
sample_rate: 22050  # 音频采样率
fft_window_shift: 256 # 提取频谱的参数，相邻两帧的偏移量（按采样点算） (相当于 librosa.stft 中的 hop_length)
fft_window_size: 1024  # 提取频谱的参数，窗函数的宽度 (相当于 librosa.stft 中的 win_length)
fft_size: 1024  # 提取频谱的参数，短时傅里叶变换中一帧包含的采样点数 (相当于 librosa.stft 中的n_fft)
mel_bands: 80  # 提取 mel 频谱的参数，mel 频谱分段数 (相当于 librosa.filters.mel 中的 n_mels)
mel_fmin: 0.0  # 提取 mel 频谱的参数，频率下限 (相当于 librosa.stft 中的 f_min)
mel_fmax: 8000.0  # 提取 mel 频谱的参数，频率上限 (相当于 librosa.stft 中的 f_max)

seed: 1234  # 随机数种子
learning_rate: 0.0002   # 学习率
batch_size: 8   # 一个 batch 的句子数
test_every: 2000   # 每隔 n 步测试
save_every: 10000  # 每隔 n 步 保存
max_iterations: 3000000   # 最大更新次数

sigma: 1.0   # 合成音频时需要先从一个高斯分布中采样，高斯分布的标准差
n_flows: 8  # 模型的 flow 个数，
n_group: 16  # waveflow 模型的折叠系数，每 n 个采样点为一组
n_layers: 8  # 每一个 flow 包含的层数，（如 8 flow, 每个 flow 8 layer, 则共 64 layer)
n_channels: 64  # waveflow 模型内部的 channel 数
kernel_h: 3  # waveflow 模型内部的卷积窗口高度
kernel_w: 3  # waveflow 模型内部的卷积窗口宽度

**Comment by iclementine** [comment]
如果需要用 thchs30 训练中文模型，可能需要知道音频的采样率，需要做一些修改，fft 相关的几个参数也需要相应得做一些更改（22050Hz 的音频 1024 个采样点约 0.046s, 如果你的采样率不同，也要调整 stft 的参数，fft_window_size 默认和 fft_size 一样，fft_window_shift 一般取 fft_size 的 1/4）。（当然你也可以选择将音频重采样到 22050Hz.）

**Comment by lfchener** [comment]
# TransformerTTS 的配置文件
audio:
  num_mels: 80 # 提取 mel 频谱的参数，mel 频谱分段数 (相当于 librosa.filters.mel 中的 n_mels)
  n_fft: 1024 # 提取频谱的参数，短时傅里叶变换中一帧包含的采样点数 (相当于 librosa.stft 中的n_fft)
  sr: 22050 # 音频采样率
  preemphasis: 0.97 # 频谱预加重的参数
  hop_length: 256 # 提取频谱的参数，相邻两帧的偏移量（按采样点算） (相当于 librosa.stft 中的 hop_length)
  win_length: 1024 # 提取频谱的参数，窗函数的宽度 (相当于 librosa.stft 中的 win_length)
  power: 1.2 # 对结果做指数的次方值
  fmin: 0 # 提取 mel 频谱的参数，频率下限 (相当于 librosa.stft 中的 f_min)
  fmax: 8000  # 提取 mel 频谱的参数，频率上限 (相当于 librosa.stft 中的 f_max)

network:
  hidden_size: 256 # 模型隐藏层的大小
  embedding_size: 512  # embedding 层的大小
  encoder_num_head: 4 # encoder 中 multihead attention 的 head 数
  encoder_n_layers: 3 # encoder 的层数
  decoder_num_head: 4 # decoder 中 multihead attention 的 head 数
  decoder_n_layers: 3 # decoder 的层数
  outputs_per_step: 1 # decoder 每步预测的帧数
  stop_loss_weight: 8 # stop token 计算 loss 时对正例样本的加权值

vocoder:
  hidden_size: 256 # cbhg 中隐藏层的大小
  
train:
  batch_size: 32 #每个 batch 的句子数
  learning_rate: 0.001 #学习率
  warm_up_step: 4000 # 学习率衰减的启动步数
  grad_clip_thresh: 1.0 #梯度裁剪的阈值

  checkpoint_interval: 1000 #隔 n 步保存一个 checkpoint
  image_interval: 2000 #隔 n 步在 log 中生成一张对角线的图片

  max_iteration: 500000 #最大更新次数

**Comment by Godlikemandyy** [comment]
> > 请问你是需要了解哪一个模型的 yaml 配置呢？
> > 嗯， 后续版本我们会考虑加上更加友好的说明。
> 
> 例如Parakeet/examples/transformer_tts/configs/ljspeech.yaml和Parakeet/examples/waveflow/configs/waveflow_ljspeech.yaml 这些yaml文件里面的参数，是否可提供说明。如果想训练中文数据集thchs30应该如何修改

请问你训练了中文数据集了吗，我也先试试
